osmUtils/utils_geo.py

```diff
@@ -0,0 +1,141 @@
+
+"""General utility functions."""
+from .settings import DEFAULT_CRS, DEFAULT_TILES
+import geopandas as gpd
+import pandas as pd
+import mercantile as mt
+from shapely.geometry import shape, MultiPolygon, Polygon
+
+def generate_tiles(crs, zoom):
+        """
+        Generate tiles for the manifest.
+        """
+
+        #generate tiles
+        tiles = []
+        
+        #calculate bbox for input geometry if geom is none:
+
+        for tile in mt.tiles(*DEFAULT_TILES, zoom, truncate=False):
+
+            tile_id = f"{tile.z}_{tile.x}_{tile.y}"
+
+            geom = mt.feature(tile)['geometry']
+            polygon = shape(geom)
+
+            tiles.append({
+                'tile_id': tile_id,
+                'geometry': polygon
+            })
+
+        # generate geodataframe with tiles
+        gdf = gpd.GeoDataFrame(tiles)
+        
+        #check projection 
+        if gdf.crs is None:
+            gdf = set_crs(gdf, crs)
```

Comment:
This line gives me an error currently when i pass a shapely geometry to `OsmCollection`

Reply:
its working ok for me.

Reply:
I think it was my version of GPD